### ISSUE TITLE
Manual data preprocess

### ISSUE THREAD

**Issue by reobroqn** [LiyingCheng95/EntityDescriptionGeneration#1]
Hello! I have read your paper. It's excellent!
I'm interested in it. And i want to try it but didn't find detail how to preprocess dataset.
I also want to try it for my own dataset. So it would be appreciate if you can let me know.
Did take a look in **convert_levi.py** but still don't know how to do it, what is levi.txt ? categories_QID.txt ?
Thanks a lot!!!

**Comment by LiyingCheng95** [comment]
Hi! Thanks for your kind compliments!
I just uploaded a file named **convert_multi_graph.py**. You can modify the format of your input-triples file to be the same as xxx-triples.txt in this folder, and then simply run **convert_multi_graph.py**.
levi_xxx.txt is just an intermediate step.
categories_QID.txt is an even earlier step. It shows the categories and their corresponding IDs that we extracted from Wikidata.

**Comment by reobroqn** [comment]
Thanks for your reply! You just save me a lot!
I've been struggling with DCGCN and LDC AMR dataset. Phew!!!!, they're scary.
Have a nice day.

**Comment by reobroqn** [comment]
Btw, how long have you trained to achieve the result in the paper? GPU?
Do you still keep trained model? I tried to train it on Colab but crash, my pc is also too weak for this.

**Comment by LiyingCheng95** [comment]
It depends on the dataset. For ENT-DESC, it took about 10-12 hrs or sometimes even longer. V100. 
Yes, I kept the models shown in the paper, but the sizes are pretty large.

**Comment by reobroqn** [comment]
What a pity!
But thanks for answering me! >.<

**Comment by reobroqn** [comment]
Hello! It's me again. Ha ha
I'm reading your script to find when the model would stop. I don't see you specify any parameter like --epoch in train.sh, but there's a stop point where the model would stop when it could not improve after N epoch(s).
So if the result keeps getting better then the training process would not stop?
I did find --min-num-epoch and --max-num-epoch, but just my curious about how it could stop when their default values are None. Have a nice day!

**Comment by LiyingCheng95** [comment]
If max-num-epoch is not specified, the training phase will be stopped when detecting the convergence of perplexity on the validation set.

**Comment by reobroqn** [comment]
Sorry for bothering you after a long time!
Do you have anyway to make the model continue training from last checkpoint (like fine-tune) ? If it could, can you somehow lend me your trained model? It is fine if you are not convenient.
Thanks for reading!
